infra/gravity/node_commands.go

```diff
@@ -587,6 +593,57 @@ func (g *gravity) RunInPlanet(ctx context.Context, cmd string, args ...string) (
 	return out, nil
 }
 
+// IsLeader returns true if node is leader, false o.w.
+func (g *gravity) IsLeader(ctx context.Context) bool {
+	// TODO: use planet cmd once planet cli is updated s.t. `leader view` does
+	// not require args.
+	// leaderIP, err := g.RunInPlanet(ctx, "planet", "leader", "view")
+	leaderIP, err := g.RunInPlanet(ctx, "etcdctl", "get", "gravity/local/leader")
+	if err == nil && leaderIP == g.Node().PrivateAddr() {
+		return true
+	}
+	return false
+}
+
+// Connect connects the node to the cluster.
+func (g *gravity) Connect(ctx context.Context) error {
+	status, err := g.Status(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	for _, node := range status.Cluster.Nodes {
+		// TODO: look into using existing libs for manipulating network traffic.
```

Comment:
coreos/go-iptables might be worth checking out. Here's an example from wormhole: https://github.com/gravitational/wormhole/blob/master/pkg/iptables/iptables.go